sdk/messaging/azservicebus/sender.go

```diff
@@ -123,7 +149,14 @@ type ScheduleAMQPAnnotatedMessagesOptions struct {
 // delivered can be cancelled using `Receiver.CancelScheduleMessage(s)`
 // If the operation fails it can return an [*azservicebus.Error] type if the failure is actionable.
 func (s *Sender) ScheduleAMQPAnnotatedMessages(ctx context.Context, messages []*AMQPAnnotatedMessage, scheduledEnqueueTime time.Time, options *ScheduleAMQPAnnotatedMessagesOptions) ([]int64, error) {
-	return scheduleMessages(ctx, s.links, s.retryOptions, messages, scheduledEnqueueTime)
+	var err error
+	ctx, endSpan := s.startSpan(ctx, "ScheduleAMQPAnnotatedMessages", tracing.ScheduleOperationName,
+		tracing.Attribute{Key: tracing.BatchMessageCount, Value: int64(len(messages))},
+	)
+	defer func() { endSpan(err) }()
```

Comment:
@jhendrixMSFT, would it be worth building this pattern (via a callback, probably) into the tracing library? It can be internal, but it seems like everyone's going to do the "last error gets passed to endSpan before block ends" pattern.

Reply:
If not, @karenychen, we can build a helper function - maybe we'd stick it right in the retry function to make things easier since we're passing very similar information to both.

Reply:
Maybe it goes in the `sdk/internal` module?

Reply:
Sounds good, I noticed the .NET SDK gives a more general naming to their spans, we can also follow their naming convention: https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-end-to-end-tracing?tabs=net-standard-sdk-2#instrumented-operations